modules/045-snapshot-controller/templates/snapshot-validation-webhook/deployment.yaml

```diff
@@ -0,0 +1,77 @@
+kind: VerticalPodAutoscaler
+metadata:
+  name: d8-snapshot-validation-webhook
+  namespace: kube-system
+  {{- include "helm_lib_module_labels" (list . (dict "app" "d8-snapshot-validation-webhook" "workload-resource-policy.deckhouse.io" "master")) | nindent 2 }}
+spec:
+  targetRef:
+    apiVersion: "apps/v1"
+    kind: Deployment
+    name: d8-snapshot-validation-webhook
+  updatePolicy:
+    updateMode: "Auto"
+{{- end }}
+---
+apiVersion: policy/v1beta1
+kind: PodDisruptionBudget
+metadata:
+  annotations:
+    helm.sh/hook: post-upgrade, post-install
+    helm.sh/hook-delete-policy: before-hook-creation
+  name: d8-snapshot-validation-webhook
+  namespace: kube-system
+  {{- include "helm_lib_module_labels" (list . (dict "app" "d8-snapshot-validation-webhook" )) | nindent 2 }}
+spec:
+  minAvailable: {{ include "helm_lib_is_ha_to_value" (list . 1 0) }}
+  selector:
+    matchLabels:
+      app: d8-snapshot-validation-webhook
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: d8-snapshot-validation-webhook
+  namespace: kube-system
+  {{- include "helm_lib_module_labels" (list . (dict "app" "d8-snapshot-validation-webhook" )) | nindent 2 }}
+spec:
+  replicas: {{ include "helm_lib_is_ha_to_value" (list . 2 1) }}
+  selector:
+    matchLabels:
+      app: d8-snapshot-validation-webhook
+  template:
+    metadata:
+      labels:
+        app: d8-snapshot-validation-webhook
+    spec:
+      imagePullSecrets:
+        - name: deckhouse-registry
+      {{- include "helm_lib_priority_class" (tuple . "cluster-low") | nindent 6 }}
+      {{- include "helm_lib_node_selector" (tuple . "master") | nindent 6 }}
+      {{- include "helm_lib_tolerations" (tuple . "master") | nindent 6 }}
+      {{- include "helm_lib_module_pod_security_context_run_as_user_nobody" . | nindent 6 }}
+      {{- include "helm_lib_pod_anti_affinity_for_ha" (list . (dict "app" "d8-snapshot-validation-webhook")) | nindent 6 }}
+      containers:
+      - name: snapshot-validation
+        {{- include "helm_lib_module_container_security_context_read_only_root_filesystem_capabilities_drop_all" . | nindent 8 }}
+        args: 
+        - --tls-cert-file=/etc/snapshot-validation-webhook/certs/tls.crt
+        - --tls-private-key-file=/etc/snapshot-validation-webhook/certs/tls.key
+        - --port=4443
+        image: {{ .Values.global.modulesImages.registry }}:{{ .Values.global.modulesImages.tags.snapshotController.snapshotValidationWebhook }}
+        imagePullPolicy: IfNotPresent
+        ports:
+        - containerPort: 4443
+        resources:
+          requests:
+            {{- include "helm_lib_module_ephemeral_storage_only_logs" . | nindent 12 }}
+        volumeMounts:
+          - name: snapshot-validation-webhook-certs
+            mountPath: /etc/snapshot-validation-webhook/certs
+            readOnly: true
+      volumes:
+        - name: snapshot-validation-webhook-certs
+          secret:
+            secretName: d8-snapshot-validation-webhook-certs
```

Comment:
Do we need to add the checksum annotation to reload pods on every secret change?